Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -637,6 +641,29 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
   PropertyWithValue();
 };
 
+#define PROPERTYWITHVALUE_SAVEPROPERTY(type)                                   \
+  template <>                                                                  \
+  inline void PropertyWithValue<type>::saveProperty(::NeXus::File *file) {     \
+    file->makeGroup(this->name(), "NXlog", 1);                                 \
+    file->writeData("value", m_value);                                         \
+    file->closeGroup();                                                        \
+  }
+
+PROPERTYWITHVALUE_SAVEPROPERTY(float)
+PROPERTYWITHVALUE_SAVEPROPERTY(double)
+PROPERTYWITHVALUE_SAVEPROPERTY(int32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(int64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::string)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<double>)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<int32_t>)
```

Comment:
@quantumsteve It might be that I did not really understand the issue with Nexus here, but could you use `std::enable_if` instead of the macros? Something like

```cpp
template <class T,
          class = typename std::enable_if<
              std::is_same<T, float>::value ||
              std::is_same<T, double>::value ||
              std::is_same<T, int32_t>::value>::type>
inline void PropertyWithValue<type>::saveProperty(::NeXus::File *file) {
  file->makeGroup(this->name(), "NXlog", 1);
  file->writeData("value", m_value);
  file->closeGroup();
}
```